base/abstractarray.jl

```diff
@@ -51,7 +51,11 @@ size{N}(x, d1::Integer, d2::Integer, dx::Vararg{Integer, N}) = (size(x, d1), siz
 
 Returns the valid range of indices for array `A` along dimension `d`.
 """
-indices{T,N}(A::AbstractArray{T,N}, d) = d <= N ? indices(A)[d] : OneTo(1)
+function indices{T,N}(A::AbstractArray{T,N}, d)
+    @_inline_meta
```

Comment:
To expand a bit: since macros run at code-loading time (not compilation time), julia has to be capable of doing everything that the `@inline` macro calls in its internal implementation. At this point, `array.jl` hasn't even been loaded yet.